samples/v1/types_config.go

```diff
@@ -64,6 +64,25 @@ type ConfigSpec struct {
 	// content but the operator will not recreate(or update) anything
 	// listed here.
 	SkippedTemplates []string `json:"skippedTemplates,omitempty" protobuf:"bytes,6,opt,name=skippedTemplates"`
+
+	// skippedHelmCharts specifies names of helm charts that should NOT be
+	// managed. Admins can use this to allow them to delete content
+	// they don’t want. They will still have to MANUALLY DELETE the
+	// content but the operator will not recreate(or update) anything
+	// listed here. Few examples of the name of helmcharts which can be skipped are
+	// 'redhat-redhat-perl-imagestreams','redhat-redhat-nodejs-imagestreams','redhat-nginx-imagestreams',
+	// 'redhat-redhat-ruby-imagestreams','redhat-redhat-python-imagestreams','redhat-redhat-php-imagestreams',
+	// 'redhat-httpd-imagestreams','redhat-redhat-dotnet-imagestreams'. Rest of the names can be obtained from
+	// openshift console --> helmcharts -->installed helmcharts. This will display the list of all the
+	// 12 helmcharts(of imagestreams)being installed by Samples Operator. The skippedHelmCharts must be a
+	// valid Kubernetes resource name. May contain only lowercase alphanumeric characters, hyphens and periods,
+	// and each period separated segment must begin and end with an alphanumeric character. It must be non-empty
+	// and at most 253 characters in length
+	// +listType=set
+	// +kubebuilder:validation:MaxItems=16
+	// +kubebuilder:validation:XValidation:rule="self.all(x, x.matches('^[a-z0-9]([a-z0-9-]*[a-z0-9])?$'))",message="skippedHelmCharts must be a valid Kubernetes resource name .May contain only lowercase alphanumeric characters, hyphens and periods, and each period separated segment must begin and end with an alphanumeric character"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, x.size() > 0 && x.size() < 254)", message="skippedHelmCharts must be non-empty and at most 253 characters in length"
+	SkippedHelmCharts []string `json:"skippedHelmCharts,omitempty" protobuf:"bytes,7,opt,name=skippedhelmCharts"`
```

Comment:
I noticed this was failing, so had a go locally, to get this to work we need a string alias so that we can impose length restrictions on the strings in the list
```
	// +kubebuilder:validation:MaxItems=16
	// +kubebuilder:validation:XValidation:rule="self.all(x, x.matches('^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'))",message="skippedHelmCharts must be a valid Kubernetes resource name. May contain only lowercase alphanumeric characters, hyphens and periods, and each period separated segment must begin and end with an alphanumeric character"
	SkippedHelmCharts []HelmChartName `json:"skippedHelmCharts,omitempty" protobuf:"bytes,7,opt,name=skippedhelmCharts"`
}

// HelmChartName is a string alias that is used to represent the name of a helm chart.
// +kubebuilder:validation:MinLength=1
// +kubebuilder:validation:MaxLength=253
type HelmChartName string
```

Can you apply this, and then add some tests to the integration test suite for some valid, and invalid name cases please

Reply:
Applied the above, can you please point me to the test suite.?